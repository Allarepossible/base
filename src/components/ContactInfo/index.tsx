import React from 'react';
import styled from 'styled-components';

import {Contact} from 'entities/contact';

interface Props {
    size: 'l' | 's';
}

export const ContactBox = styled.div`
    display: flex;
    flex-direction: column;
`;

export const StyledSpan = styled.span`
    font-size: ${({size}: Props) => size === 'l' ? '16px' : '12px'};
`;

const ContactInfo = ({
    name,
    phone,
}: Contact) => (
    <div>
        <ContactBox>
            <StyledSpan size="s">Имя: {name}</StyledSpan>
            <StyledSpan size="s">Тел.: {phone}</StyledSpan>
        </ContactBox>
    </div>
);

export default ContactInfo;
