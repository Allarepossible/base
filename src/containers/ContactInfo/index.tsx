import {connect} from 'react-redux';

import {State, ContactId} from 'entities/contact';
import ContactInfo from 'components/ContactInfo';

interface Props {
    contactId: ContactId;
}

export const mapStateToProps = ({contact}: State, {contactId}: Props) => {
    const {name, phone, id} = contact.data[contactId] || {};

    return {name, phone, id};
};

export default connect(mapStateToProps)(ContactInfo);
