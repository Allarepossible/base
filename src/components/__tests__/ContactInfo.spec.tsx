import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import ContactInfo, {ContactBox} from '../ContactInfo';

it('Renders correct contact styles', () => {
    const result = renderer.create(<ContactBox>text</ContactBox>).toJSON();

    expect(result).toMatchSnapshot();
});

it('Renders ContactInfo', () => {
    const result = renderer.create(
        <ContactInfo
            id={1}
            name="name"
            phone="phoneNumber"
        />).toJSON();

    expect(result).toMatchSnapshot();
});
