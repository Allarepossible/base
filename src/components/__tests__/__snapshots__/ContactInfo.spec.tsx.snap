// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders ContactInfo 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  font-size: 12px;
}

<div>
  <div
    className="c0"
  >
    <span
      className="c1"
      size="s"
    >
      Имя: 
      name
    </span>
    <span
      className="c1"
      size="s"
    >
      Тел.: 
      phoneNumber
    </span>
  </div>
</div>
`;

exports[`Renders correct contact styles 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
>
  text
</div>
`;
